libgalois/include/katana/PropertyGraph.h

```diff
@@ -198,8 +198,9 @@ class KATANA_EXPORT PropertyGraph {
     uint64_t ApproxMemUse() const { return ropv.ApproxMemUse(); }
 
     Result<void> AddProperties(
-        const std::shared_ptr<arrow::Table>& props) const {
-      return (g->*add_properties_fn)(props);
+        const std::shared_ptr<arrow::Table>& props,
+        tsuba::TxnContext* txn_ctx) const {
```

Comment:
You have the transaction context as the last argument here and the first for the analytics routines. This should be consistent. I think I prefer last, since if we were doing this in a modern language this argument wouldn't exist at all, so hiding it is good.

Reply:
The convention should be that TxnContext is the last argument; though, I think we will have some exceptions when the function already takes variable arguments like

https://github.com/KatanaGraph/katana/pull/727/files#diff-576d16ef31b9271d43238ea8dfe5c19a7cd07e1761e825554316b14fff165875R97-R98

Reply:
@arthurp I prefer putting it last. But there are some cases where it cannot be put at the end. For example, some other function parameters having default values, or parameter packs as in the link posted by Donald above.

Reply:
My original thought was that something named `...Context` conventionally tends to be put as either first or last argument. But now I've changed all of the `TxnContext`-s to be last possible argument.